src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -659,16 +660,16 @@ var ReactCompositeComponentMixin = {
         transaction,
         this._context
       );
-    }
-
-    if (this._pendingStateQueue !== null || this._pendingForceUpdate) {
+    } else if (this._pendingStateQueue !== null || this._pendingForceUpdate) {
       this.updateComponent(
         transaction,
         this._currentElement,
         this._currentElement,
         this._context,
         this._context
       );
+    } else {
+      this._updateBatchNumber = null;
```

Comment:
That is, enqueueCallback marks the component dirty which is harmless when done in conjunction with enqueueSetState but kinda silly when it can get split into a separate batch.